src/profile-logic/processed-profile-versioning.js

```diff
@@ -646,5 +646,48 @@ const _upgraders = {
       }
     }
   },
+  [16]: profile => {
+    // type field on some markers were missing. Renamed category field of
+    // VsyncTimestamp and LayerTranslation marker payloads to type and added
+    // a type field to Screenshot marker payload.
+    for (const thread of profile.threads) {
+      const { stringArray, markers } = thread;
+      const stringTable = new UniqueStringArray(stringArray);
+      const newDataArray = [];
+      for (let i = 0; i < markers.length; i++) {
+        const name = stringTable.getString(markers.name[i]);
+        const data = markers.data[i];
+        switch (name) {
+          case 'VsyncTimestamp':
+            newDataArray[i] = {
+              type: 'VsyncTimestamp',
+              vsync: data.vsync,
+            };
+            break;
+          case 'LayerTranslation':
+            newDataArray[i] = {
+              type: 'LayerTranslation',
+              layer: data.layer,
+              x: data.x,
+              y: data.y,
+            };
+            break;
+          case 'Screenshot':
```

Comment:
I believe this is `CompositorScreenshot`. I'll leave another comment in the Gecko review about this.

I would say either:

A. Keep the name as `CompositorScreenshot`,
B. Upgrade it to `Screenshot` in both the name and type, and replace any other instances of `"CompositorScreenshot"` in perf.html

In Gecko: https://searchfox.org/mozilla-central/source/gfx/layers/ProfilerScreenshots.cpp#107

In perf.html: https://github.com/devtools-html/perf.html/search?q=CompositorScreenshot&unscoped_q=CompositorScreenshot